services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/migration/DatasetBackFillFileSummary.java

```diff
@@ -0,0 +1,222 @@
+
+@TemporaryCode(author = "sandhra.sokhal@sagebase.org", comment = "This allows back filling of field checksum,size and count for dataset")
+@Service
+public class DatasetBackFillFileSummary {
+    private static final long PAGE_SIZE_LIMIT = 1000;
+    private static final String CHECKSUM = "checksum";
+    private static final String SIZE = "size";
+    private static final String COUNT = "count";
+    private static final String SQL_UPDATE_ETAG = "UPDATE " + TABLE_NODE + " SET " + COL_NODE_ETAG
+            + " = ? WHERE " + COL_NODE_ID + " = ?";
+    private static final String SQL_UPDATE_ANNOTATION = "UPDATE " + TABLE_REVISION + " SET " + COL_REVISION_ENTITY_PROPERTY_ANNOTATIONS_BLOB
+            + " = ? WHERE " + COL_REVISION_OWNER_NODE + " = ? AND " + COL_REVISION_NUMBER + " =?";
+    private static final String SELECT_ALL_DATASET = "SELECT N.ID, R.NUMBER, R.ENTITY_PROPERTY_ANNOTATIONS, R.ITEMS FROM " + TABLE_NODE + " N JOIN " + TABLE_REVISION + " R ON N."
+            + COL_NODE_ID + " = R." + COL_REVISION_OWNER_NODE + " WHERE N." + COL_NODE_TYPE + " ='dataset' ORDER BY N.ID, R.NUMBER LIMIT ? OFFSET ?";
+    private static final RowMapper<DatasetBackFillDTO> DATASET_BACK_FILL_ROW_MAPPER = (rs, rowNum) -> {
+        String id = rs.getString(COL_NODE_ID);
+        long version = rs.getLong(COL_REVISION_NUMBER);
+        org.sagebionetworks.repo.model.Annotations entityPropertyAnnotation = null;
+        List<EntityRef> items;
+        byte[] bytes = rs.getBytes(COL_REVISION_ENTITY_PROPERTY_ANNOTATIONS_BLOB);
+
+        try {
+            if (bytes != null) {
+                entityPropertyAnnotation = AnnotationUtils.decompressedAnnotationsV1(bytes);
+            }
+
+            items = NodeUtils.readJsonToItems(rs.getString(COL_REVISION_ITEMS));
+            if(items == null){
+                items = Collections.emptyList();
+            }
+        } catch (IOException e) {
+            throw new DatastoreException(e);
+        }
+
+        return new DatasetBackFillDTO(id, version, entityPropertyAnnotation, items);
+    };
+    static private Logger log = LogManager.getLogger(DatasetBackFillFileSummary.class);
+    private JdbcTemplate jdbcTemplate;
+
+    private NodeDAO nodeDAO;
+
+    private TransactionTemplate readCommitedTransactionTemplate;
+
+    @Autowired
+    public DatasetBackFillFileSummary(JdbcTemplate jdbcTemplate, NodeDAO nodeDAO, TransactionTemplate readCommitedTransactionTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+        this.nodeDAO = nodeDAO;
+        this.readCommitedTransactionTemplate = readCommitedTransactionTemplate;
+    }
+
+    public List<DatasetBackFillDTO> getAllDatasetEntity(long limit, long offset) {
+        return jdbcTemplate.query(SELECT_ALL_DATASET, DATASET_BACK_FILL_ROW_MAPPER, limit, offset);
+    }
+
+    public void updateNodeEtag(String etag, String nodId) {
+        jdbcTemplate.update(SQL_UPDATE_ETAG, etag, nodId);
```

Comment:
looking at NodeDaoImpl.touch() we also send a message to the workers to notify of the etag change.  We think it would be better to do that here as well.